File: src/main/java/de/pflugradts/passbird/application/commandhandling/command/AbstractFilenameCommand.java
package de.pflugradts.passbird.application.commandhandling.command;

import de.pflugradts.passbird.domain.model.transfer.Bytes;
import de.pflugradts.passbird.domain.model.transfer.Input;

public abstract class AbstractFilenameCommand extends AbstractInputCommand {

    protected AbstractFilenameCommand(final Input input) {
        super(input);
    }

    @Override
    public Bytes getArgument() {
        return getInput().getBytes().slice(1);
    }

}

File: src/main/kotlin/de/pflugradts/passbird/application/commandhandling/command/base/AbstractFilenameCommand.kt
package de.pflugradts.passbird.application.commandhandling.command.base

import de.pflugradts.passbird.domain.model.transfer.Input

abstract class AbstractFilenameCommand(private val input: Input) : AbstractInputCommand(input) {
    override val argument get() = input.shell.slice(1)
}
